config/accounts.go

```diff
@@ -39,20 +44,147 @@ func (a *AccountCCPA) EnabledForIntegrationType(integrationType IntegrationType)
+}
+
+// FeatureOneVendorException checks if the given bidder is a vendor exception.
+func (a *AccountGDPR) FeatureOneVendorException(bidder openrtb_ext.BidderName) (value, exists bool) {
+	if a.SpecialFeature1.VendorExceptionMap == nil {
+		return false, false
+	}
+	_, exists = a.SpecialFeature1.VendorExceptionMap[bidder]
+
+	return exists, true
+}
+
+// PurposeOneTreatmentEnabled gets the account level purpose one treatment enabled setting returning the value and
+// whether or not it is set. If not set, a default value of true is returned matching host default behavior.
+func (a *AccountGDPR) PurposeOneTreatmentEnabled() (value, exists bool) {
+	if a.PurposeOneTreatment.Enabled == nil {
+		return true, false
+	}
+	return *a.PurposeOneTreatment.Enabled, true
+}
+
+// PurposeOneTreatmentAccessAllowed gets the account level purpose one treatment access allowed setting returning the
+// value and whether or not it is set. If not set, a default value of true is returned matching host default behavior.
+func (a *AccountGDPR) PurposeOneTreatmentAccessAllowed() (value, exists bool) {
+	// default to allowed matching host default behavior
+	if a.PurposeOneTreatment.AccessAllowed == nil {
+		return true, false
+	}
+	return *a.PurposeOneTreatment.AccessAllowed, true
+}
+
+// BasicEnforcementVendor checks if the given bidder is considered a basic enforcement vendor which indicates whether
+// weak vendor enforcement applies to that bidder.
+func (a *AccountGDPR) BasicEnforcementVendor(bidder openrtb_ext.BidderName) (value, exists bool) {
+	if a.BasicEnforcementVendorsMap == nil {
+		return false, false
+	}
+	_, exists = a.BasicEnforcementVendorsMap[string(bidder)]
+
+	return exists, true
+}
+
+// AccountGDPRPurpose represents account-specific GDPR purpose configuration
+type AccountGDPRPurpose struct {
+	EnforcePurpose string `mapstructure:"enforce_purpose" json:"enforce_purpose,omitempty"`
+	EnforceVendors *bool  `mapstructure:"enforce_vendors" json:"enforce_vendors,omitempty"`
+	// Array of vendor exceptions that is used to create the hash table VendorExceptionMap so vendor names can be instantly accessed
+	VendorExceptions   []openrtb_ext.BidderName `mapstructure:"vendor_exceptions" json:"vendor_exceptions"`
+	VendorExceptionMap map[openrtb_ext.BidderName]struct{}
+}
+
+// AccountGDPRSpecialFeature represents account-specific GDPR special feature configuration
+type AccountGDPRSpecialFeature struct {
+	Enforce *bool `mapstructure:"enforce" json:"enforce"`
+	// Array of vendor exceptions that is used to create the hash table VendorExceptionMap so vendor names can be instantly accessed
+	VendorExceptions   []openrtb_ext.BidderName `mapstructure:"vendor_exceptions" json:"vendor_exceptions"`
+	VendorExceptionMap map[openrtb_ext.BidderName]struct{}
+}
+
+// AccountGDPRPurposeOneTreatment represents account-specific GDPR purpose one treatment configuration
+type AccountGDPRPurposeOneTreatment struct {
+	Enabled       *bool `mapstructure:"enabled"`
```

Comment:
I am a little worried that Purpose one treatment would seem to default to off unless explicitly enabled in the account config. Is this really what we want to happen?

Reply:
Nevermind, I see they are bool pointers now.